eclass/toolchain.eclass

```diff
@@ -1,14 +1,36 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-# Maintainer: Toolchain Ninjas <toolchain@gentoo.org>
+# @ECLASS: toolchain.eclass
+# @MAINTAINER:
+# Toolchain Ninjas <toolchain@gentoo.org>
 # @SUPPORTED_EAPIS: 5 6 7
+# @BLURB: Functions to build sys-devel/gcc
+# @DESCRIPTION:
+# Comprehensive helper and phase functions to build sys-devel/gcc and
+# adjacent packages, support for release and live ebuilds.
+#
+# This eclass unconditionally inherits toolchain-funcs.eclass and all its public
+# variables and helper functions may be considered as part of this eclass's API.
+#
+# This eclass's phase functions are not intended to be mixed and matched, so if
+# any phase functions are overridden the version here should also be called.
+
+case ${EAPI:-0} in
+	0|1|2|3|4*) die "Need to upgrade to at least EAPI=5" ;;
```

Comment:
We never approved 4-*, 5-*, so Iw wouldn't worry too much about those, but up to you. I'd go for "I don't speak" in those cases as it's unclear if we could even support them properly in the past.

Reply:
didn't want to change the existing logic, maybe we get a second opinion through dev ML review...

Reply:
`4*` will match EAPI 40 :smile: so please don't use it.

Reply:
hah.